crates/bevy_transform/src/systems.rs

```diff
@@ -165,6 +183,59 @@ mod test {
     use crate::TransformBundle;
     use bevy_hierarchy::{BuildChildren, BuildWorldChildren, Children, Parent};
 
+    #[test]
+    fn correct_parent_removed() {
+        ComputeTaskPool::init(TaskPool::default);
+        let mut world = World::default();
+        let offset_global_transform =
+            |offset| GlobalTransform::from(Transform::from_xyz(offset, offset, offset));
+        let offset_transform =
+            |offset| TransformBundle::from_transform(Transform::from_xyz(offset, offset, offset));
+
+        let mut schedule = Schedule::new();
+        schedule.add_system(sync_simple_transforms);
+        schedule.add_system(propagate_transforms);
+
+        let mut command_queue = CommandQueue::default();
+        let mut commands = Commands::new(&mut command_queue, &world);
+        let root = commands.spawn(offset_transform(3.3)).id();
+        let parent = commands.spawn(offset_transform(4.4)).id();
+        let child = commands.spawn(offset_transform(5.5)).id();
+        commands.entity(parent).set_parent(root);
+        commands.entity(child).set_parent(parent);
+        command_queue.apply(&mut world);
+        schedule.run(&mut world);
+
+        assert_ne!(
+            world.get::<GlobalTransform>(parent).unwrap(),
+            &GlobalTransform::from(Transform::IDENTITY)
+        );
```

Comment:
I don't understand why this check passes. This is after transform propagation has been run for the first time, but before `parent`'s parent has been removed. Shouldn't `parent`'s translation be `3.3 + 4.4`?

Reply:
This tests that when I run `schedule.run(&mut world)`, the `GlobalTransform` is indeed updated (ie: the propagation systems have run) so that the following assertions are not just an accident due to transform never changing. I had to think about this a bit, it is indeed really weird.